### ISSUE TITLE
Missing schema definitions

### ISSUE THREAD

**Issue by rlilov** [dcsaorg/DCSA-OpenAPI#15]
Hello,
Reading TnT Interface standard document https://dcsa.org/data-and-interfaces/interface-standard-for-track-and-trace-v1-2/
I can't seem to find some important concepts from it implemented in the actual API
Namely schema for  Transport Reference, Transport Leg Reference, Facility (in my understanding this is location where event takes place).
I have been trying to implement this API (as provider)  for one of my products, but seem to be stuck on those missing properties.

Kindly advise
Thanks
Rumen

**Comment by AsgerStage** [comment]
Hi Rumen,

Apologies for the delay, I just got back from vacation.

I assume this is for Track & Trace 1.2, as the document you linked.

Is this what you are looking for: https://github.com/dcsaorg/DCSA-OpenAPI/blob/master/tnt/v1/tnt.yaml#L288 ? From line 288 to 293, the 3 schemas, that you mentioned, are referenced.

Hope that helps, but let me know if there is anything else I can do.

Best,
Asger

**Comment by rlilov** [comment]
Hi Asger,
I hope you had a nice holidays.
My API discovery process is as follows:
1. starting form DCSA page I visited TnT section https://dcsa.org/initiatives/info-model-and-track-and-trace/
2. after reading carefully the published PDF documents I navigate out to published SwaggerHub link https://app.swaggerhub.com/apis/dcsaorg/DCSA_TNT/2.0.0 and try to study schemas, generate code via Export/Server stub link. 
In this process, I am not observing definitions to above mentioned schemas.
n.b.: At no point of time I felt the need to directly consult yaml files.
I would assume this would be the API discovery path for most of potential adopters.

The event schema returned object by itself, does not contain them and instead contains some strange (at least for me) properties like: "transportCallID","delayReasonCode", "vesselScheduleChangeRemark" that seem more related to eventual schedules payload.

Kindly advise if I am missing something and where I can find url, where those schemas will be documented via OpenAPI doc and code.
Thanks
Rumen


Thnaks
Rumen

**Comment by AsgerStage** [comment]
Hi Rumen,

Thanks for the detailed explanation, I understand the problem now.

The link from your point 1, to the dcsa.org site, links to version 2.0.0 of the API where the API uses Transport Calls for location. The link to version 1.2.0, that I think you are looking for, is here: https://app.swaggerhub.com/apis/dcsaorg/DCSA_TNT/1.2.0#/ 

This link can also be found via SwaggerHub like this: 
![image](https://user-images.githubusercontent.com/15195163/91700937-df1b5480-eb76-11ea-95e6-0f810a55d481.png)

Using that page, you should be able to find the information you need via the Swaggerhub GUI.

As a sidenote: 2.0 can also be implemented, but as you noticed, locations are implemented in a different way.

I will forward your feedback on API discovery to my teammates, as I think you made some very valuable observations here.

Best,
Asger

**Comment by rlilov** [comment]
Thanks Asger,
I was incorrectly assuming 2.0.0 version is the one to take.
Looking now at 1.2.0 and please don't take it as a critique, simply constructive suggestion, while I hope I am looking at the right definition.  I was hoping to see Location (Facility) object described like in rich schema, encapsulating optionally  UN code, address, coordinates, Google Plus code (why not) etc. What I saw is facilityTypeCode defined as string, I would have expected it to be minimum well defined enum.
Where I am coming from, is application that reports container movements, while in inland. Customer premises are often involved, and they are rarely (if ever) UN codified. I was needing ability to report container locations within the boundaries of well defined address and/or coordinates. Carriers, for one, can appreciate this data (I hope).

Now that I have viability in 1.20 and 2.0.0, I am not sure I understand the direction taken in 2.0.0 , hope it is just me.
Confusion comes from reading 1 issue below mine, where trasnportCallId looks like a candidate for removal, so I was wondering how 2.0.0 will look location-wise.

Anyway, you can close this, as you answered my question, where the schema are defined.
I tried to give some feedback on how they are defined, not sure how useful it is.

Best regards
Rumen

**Comment by AsgerStage** [comment]
Hi Rumen,

Glad to hear it was sorted.

Regarding your comments on how locations work in 2.0, I can't explain it very in-depth, as I am not very involved in the data model design.  Transport Calls are not a candidate for removal, but in #12 as you referenced, Transport Call IDs were only removed from Shipment Events (Where they don't belong), but not from transport and equipment events.

In the future, we have discussed and will most likely be adding a location API, that the different DCSA APIs will use to add location information to objects. I think it is safe to assume that we will also be looking at making the schemas more descriptive while adding this API.

Closing this issue, but feel free to reach out if needed.

Best,
Asger
